### DISCUSSION TITLE
v0.12.7

### DISCUSSION THREAD

**Discussion by AsamK** [AsamK/signal-cli#1400] [Releases]
**Full Changelog**: https://github.com/AsamK/signal-cli/compare/v0.12.6...v0.12.7

<hr /><em>This discussion was created from the release <a href='https://github.com/AsamK/signal-cli/releases/tag/v0.12.7'>v0.12.7</a>.</em>

**Comment by bbernhard** [comment]
Thanks a lot for the quick release and the fix for the device linking - very much appreciated!

Just a quick question regarding the libsignal-client version: I think the newest version now requires 0.36.1, right? (at least I've seen that libsignal-client 0.36.1 is now bundled with the released version)

**Reply by AsamK** [reply]
Yes, indeed, I wanted to add that to the CHANGELOG, but forgot ...

**Reply by bbernhard** [reply]
No worries!

Many thanks! :)

**Reply by m-ueberall** [reply]
Luckily, the local build script inspects the upstream archive and triggers library build runs accordingly… :grin:

**Comment by pmatuszy** [comment]
there is a problem during the compilation on ubuntu:

```
cargo build -p libsignal-jni --release
   Compiling boring-sys v3.1.0 (https://github.com/signalapp/boring?branch=libsignal#8245063a)
   Compiling pqcrypto-internals v0.2.5
   Compiling pqcrypto-kyber v0.7.6
   Compiling pqcrypto-kyber v0.8.0
   Compiling num-integer v0.1.45
   Compiling snow v0.9.3
   Compiling libsignal-protocol v0.1.0 (/tmp/signal-cli-install/libsignal/rust/protocol)
error: failed to run custom build command for `libsignal-protocol v0.1.0 (/tmp/signal-cli-install/libsignal/rust/protocol)`

Caused by:
  process didn't exit successfully: `/tmp/signal-cli-install/libsignal/target/release/build/libsignal-protocol-718afc7d65a4d7bd/build-script-build` (exit status: 101)
  --- stderr
  thread 'main' panicked at rust/protocol/build.rs:18:10:
  Protobufs in src are valid: Custom { kind: Other, error: "protoc failed: Unknown flag: --experimental_allow_proto3_optional\n" }
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...

```

i have the following versions:

```
 # cargo --version
cargo 1.74.0-nightly (96fe1c9e1 2023-08-29)
 # rustc --version
rustc 1.74.0-nightly (2f5df8a94 2023-08-31)
```

**Reply by bbernhard** [reply]
If I remember correctly, the issue is `protoc`, which is too old. Try to upgrade it.

**Reply by pmatuszy** [reply]
thank you very much for your QUICK response. Just for future reference - install the latest version for your architecture from here:
https://github.com/protocolbuffers/protobuf/releases

**Comment by pmatuszy** [comment]
Hi 

from version 0.12.1 there is no longer build just for Linux:
signal-cli-0.12.7-Linux.tar.gz   (withouht -native)

Is it possible to have it built?

**Reply by m-ueberall** [reply]
> from version 0.12.1 there is no longer build just for Linux: `signal-cli-0.12.7-Linux.tar.gz` (without `-native`)

From version 0.12.3 onwards, the `signal-cli-0.12.n-Linux.tar.gz` and `signal-cli-0.12.n-Windows.tar.gz` archives have been combined (now named `signal-cli-0.12.n.tar.gz`). The included `…/lib/libsignal-client-0.aa.b.jar` contains the required `.so` library for Linux as well as its `.dll` counterpart for Windows:
```
[2024-01-18T23:46:47+0100] ueberall_l@desktop01:/tmp% unzip -Z signal-cli-0.12.3/lib/libsignal-client-0.32.1.jar| grep -E '.so|.dll'
-rwxr-xr-x  2.0 unx 26690280 b- defN 23-Sep-20 19:55 libsignal_jni.so
-rw-r--r--  2.0 unx  3471872 b- defN 23-Sep-20 19:55 signal_jni.dll
```
